### ISSUE TITLE
Completely useless post appearing at the top of timelines in the web client

### ISSUE THREAD

**Issue by hollerith** [mastodon/mastodon#25957]
### Steps to reproduce the problem

1. Visit your home timeline
2. That's it, maybe you see it maybe you don't: intermittent

### Expected behaviour

Not having this post 

### Actual behaviour

Post appears intermittently

### Detailed description

There is a post which is not real. 

It says "This is your home base within Mastodon". It appears at the top of your timeline. It serves no purpose whatsoever. It is extra annoying because it cannot be disabled and it appears intermittently. 

### Specifications

Brave on Linux
Mastodon 4.1.4+nightly-20230711

**Comment by ClearlyClaire** [comment]
I assume you are talking about the following banner?

![image](https://github.com/mastodon/mastodon/assets/384364/a71dc6b3-e361-4048-ae97-b442690e7bcb)

> It serves no purpose whatsoever.

It's used to guide people to the explore feature, as some people complain that their timeline is too slow.

> It is extra annoying because it cannot be disabled and it appears intermittently.

Clicking the “x” will dismiss it and it won't come back.

**Comment by trwnh** [comment]
Maybe changing the "x" to a proper Dismiss button?

**Comment by Gargron** [comment]
The "x" button is pretty widespread.

**Comment by legoktm** [comment]
> Clicking the “x” will dismiss it and it won't come back.

I've seen complaints that this isn't reliable and it does come back ([example](https://bikeshed.vibber.net/@brion/111105999595010301)).

If I traced the code properly, it's storing dismissal state in localStorage (c.f. [settings.js](https://github.com/mastodon/mastodon/blob/4fcc026f0f1b12a9de21a3af33375a9c8867dd55/app/javascript/mastodon/settings.js)) - is that correct?

localStorage isn't that persistent, it's per device, and people often have browser privacy settings/plugins that clear it regularly. I would suggest storing dismissal state server-side if it really should be a "dismiss once and never see it again" thing. Please let me know if I should file a separate task for this.

**Comment by brendanjones** [comment]
Came here to report that I keep getting this reappearing on desktop. I've dismissed it but it comes back. 

I don't know the technical details but when this banner was released I think I remember reading that it appears when there aren't many posts in your feed? This would match the fact that I *think* the banner appears when there's a delay in my page loading, like when my VPN is reconnecting. Is it possible that the banner is triggered before the feed content loads? In which case, change the banner check to occur only after feed content is fetched? (I am not a dev, so take this with a grain of salt)
